test/native/helpers.js

```diff
@@ -35,35 +35,39 @@ provideToNativeHtml.mockImplementation( ( html ) => {
 } );
 
 export function initializeEditor( props ) {
-	const renderResult = render(
+	const screen = render(
 		<Editor
 			postId={ `post-id-${ uuid() }` }
 			postType="post"
 			initialTitle="test"
 			{ ...props }
 		/>
 	);
-	const { getByTestId } = renderResult;
+	const { getByTestId } = screen;
 
 	// A promise is used here, instead of making the function async, to prevent
 	// the React Native testing library from warning of potential undesired React state updates
 	// that can be covered in the integration tests.
 	// Reference: https://git.io/JPHn6
 	return new Promise( ( resolve ) => {
-		waitFor( () => getByTestId( 'block-list-wrapper' ) ).then(
-			( blockListWrapper ) => {
-				// onLayout event has to be explicitly dispatched in BlockList component,
-				// otherwise the inner blocks are not rendered.
-				fireEvent( blockListWrapper, 'layout', {
-					nativeEvent: {
-						layout: {
-							width: 100,
-						},
+		// Some of the store updates that happen upon editor initialization are executed at the end of the current
+		// Javascript block execution and after the test is finished. In order to prevent "act" warnings due to
+		// this behavior, we wait for the execution block to be finished before acting on the test.
+		act(
+			() => new Promise( ( actResolve ) => setImmediate( actResolve ) )
+		).then( () => {
```

Comment:
I totally understand your point, I'm actually gathering insights for providing a full context, I'll let you know when it's ready 🙇 .

Reply:
@dcalhoun I added a `Context` section in the PR's description regarding this topic, it's a bit hard to explain as it's related to asynchronicity and store updates, so let me know if you'd like me to expand anything.

Reply:
> @dcalhoun I added a `Context` section in the PR's description regarding this topic, it's a bit hard to explain as it's related to asynchronicity and store updates, so let me know if you'd like me to expand anything.

Fantastic! Thank you _so much_ for taking the time to document this context. It is immensely helpful for me, and I imagine it will be for future readers as well. 

> Probably, the best way to have a better context is by modifying the code and adding some breakpoints, I referenced where I placed mine for testing ([reference](https://github.com/WordPress/gutenberg/blob/e1978ea1bb20e99be9f64a189ac5f898a7cee594/packages/data/src/redux-store/index.js#L415)) in case it helps out, but feel free to test further in this regard.

Agreed. I will likely circle back to further explore this subject at some point. Specifically, thanks to the wonderful context you provided, I wonder if we might be able to wrap either the [resolver promise](https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#an-alternative-waiting-for-the-mocked-promise) and/or a [manual tick of fake timers](https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#other-use-cases-for-manually-calling-act) with `act` to acutely await the core asynchronous work. I.e. I question if `Promise` + `setImmediate` is the most straightforward way of expressing what we are awaiting. 

I do not share that thought to question the current implementation or imply that we should explore it now, but merely "thinking out loud."